subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
```

Comment:
If the interface is already set for the context, then this should not be done. If the app / other part of the stack has already set the interface, this code now overwrites that value.